content/chapters/io/lab/content/file-descriptors.md

```diff
@@ -70,14 +70,13 @@ It returns the number of bytes that were read from the file.
 
 Notice that the code doesn't simply call `read(fd, buff, bytes_to_read)`.
 Instead, it uses a `while` loop to read data from the file.
-The reason is, as always, stated in [`read()`'s man page](https://man7.org/linux/man-pages/man2/read.2.html).
-The snippet below is about the number of bytes returned by `read()`:
 
-> It is not an error if this number is smaller than the number of
-> bytes requested; this may happen for example because fewer bytes
-> are actually available right now (maybe because we were close to
-> end-of-file, or because we are reading from a pipe, or from a
-> terminal), or because read() was interrupted by a signal.
+The return value of `read()` may differ from _bytes_to_read_ if there are not enough bytes available or if the operation is interrupted by a signal.
+A return value between **0** and _bytes_to_read_ is not enough to decide whether we should stop reading.
+To determine this, we make another `read()` call, which will return **0** if the cursor is already at **EOF** (end of file).
+
```

Comment:
I would also add a link to [the "RETURN VALUE" section from `read()`'s man page](https://man7.org/linux/man-pages/man2/read.2.html#RETURN_VALUE)